### DISCUSSION TITLE
Line break option when using multiple conditional expressions

### DISCUSSION THREAD

**Discussion by vsfeedback** [dotnet/roslyn#46602] [General]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/idea/1133304/line-break-option-when-using-multiple-conditional.html)._

---
The IDE0046 suggestion makes a series of multiple conditional expressions awkward.

----

Original Code:
```
        static int Foo(bool f1, bool f2, bool f3)
        {
            if (f1)
            {
                return 123;
            }
            if (f2)
            {
                return 456;
            }
            if (f3)
            {
                return 789;
            }
            return 0;
        }
```

----

Applied IDE0046  suggestion:
```
        static int Foo(bool f1, bool f2, bool f3)
        {
            return f1 ? 123 : f2 ? 456 : f3 ? 789 : 0;
        }
```

----

I want the code formatted to be cleanly formatted, with colon followed by a new line of code.

```
        static int Foo2(bool f1, bool f2, bool f3)
        {
            return f1 ? 123 :
                f2 ? 456 :
                f3 ? 789 :
                0;
        }
```

---
#### Feedback Bot on 8/3/2020, 03:18 AM: 

Thank you for taking the time to provide your suggestion.&nbsp; We will do some preliminary checks to make sure we can proceed further.&nbsp; We'll provide an update once the issue has been triaged by the product team.

**Comment by CyrusNajmabadi** [comment]
IDE will already wrap this if the line is long (i think that is measured at about 120 cols).  Wrapping uses the nice form of:

```
return f1
  ? 123
  : f2...
```

That way when-true/when-false clauses are aligned, which makes it clearer what is paired with what.
